src/sheets/quadrone/Tidy5eCharacterSheetQuadrone.svelte.ts

```diff
@@ -328,6 +313,39 @@ export class Tidy5eCharacterSheetQuadrone extends Tidy5eActorSheetQuadroneBase(
     return context;
   }
 
+  private async _getPortrait(): Promise<
+    CharacterSheetQuadroneContext['portrait']
+  > {
+    const { actor } = this;
+    const defaults = Actor.implementation.getDefaultArtwork(actor._source);
+    const themeSettings = ThemeQuadrone.getSheetThemeSettings({ doc: actor });
+    const showToken =
+      actor.flags.dnd5e?.[CONSTANTS.SYSTEM_FLAG_SHOW_TOKEN_PORTRAIT] === true ||
+      themeSettings.portraitShape === 'token';
+    const effectiveToken = actor.isToken ? actor.token : actor.prototypeToken;
+    const isRandom = !!effectiveToken?.randomImg;
+    const rawSrc = showToken
+      ? effectiveToken?.texture.src ?? actor.img
+      : actor.img;
+
+    let src = rawSrc;
+    if (showToken && isRandom) src = randomItem(await actor.getTokenImages());
+
+    src = src?.trim();
+    src ||= showToken ? defaults.texture.src : defaults.img;
+
+    const isVideo = FoundryAdapter.hasVideoExtension(src);
+
+    return {
+      src,
+      path: showToken ? 'prototypeToken.texture.src' : 'img',
+      shape: showToken ? 'token' : themeSettings.portraitShape ?? 'round',
+      isVideo,
+      isRandom,
+      truePath: isRandom ? rawSrc : undefined,
```

Comment:
I've added these because when inserting the image, any change tries to write the token image again, which overrides the randomToken with a specific one. If you have a better solution, I'd be happy to change

Reply:
Found those steps. Foundry automatically doing stuff!